lib/greenhouse_api/client.rb

```diff
@@ -1,169 +1,52 @@
-# typed: true
+# typed: false
 # frozen_string_literal: true
 
-require 'faraday'
-require 'base64'
-require 'json'
-require 'sorbet-runtime'
-
 module GreenhouseApi
-  class Response
-    attr_reader :headers, :body, :status
-
-    def initialize(headers:, body:, status:)
-      @headers = headers
-      @body = body
-      @status = status
-    end
-  end
-
   class Client
-    extend T::Sig
-
-    MAX_PER_PAGE = 500
-    ON_BEHALF_OF = 'On-Behalf-Of'
-    API_URL = 'https://harvest.greenhouse.io'
-
     def initialize(api_key)
       @api_key = api_key
     end
 
+    def list_many(resource, params = {})
+      base_client.list_many(resource, params)
+    end
+    
     def list_candidates(params = {})
-      list_many('candidates', params)
+      candidates_client.list_all(params)
```

Comment:
I kinda like your previous namespaced approach `client.candidates.all` but it also introduces a breaking change, so feel free to keep this flatten interface approach. No strong opinion from me